packages/@webex/internal-plugin-metrics/src/new-metrics.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable valid-jsdoc */
+
+// @ts-ignore
+import {WebexPlugin} from '@webex/webex-core';
+
+import CallDiagnosticMetrics, {
+  SubmitClientEventOptions,
+} from './call-diagnostic/call-diagnostic-metrics';
+import {
+  RecursivePartial,
+  ClientEvent,
+  FeatureEvent,
+  BehavioralEvent,
+  OperationalEvent,
+  MediaQualityEvent,
+} from './metrics.types';
+
+/**
+ * Metrics plugin to centralize all types of metrics.
+ * @class
+ */
+class Metrics extends WebexPlugin {
+  // eslint-disable-next-line no-use-before-define
+  static instance: Metrics;
+
+  // Helper classes to handle the different types of metrics
+  callDiagnosticMetrics: CallDiagnosticMetrics;
+
+  /**
+   * Constructor
+   * @param args
+   * @constructor
+   * @private
+   * @returns
+   */
+  constructor(...args) {
+    super(...args);
+
+    if (!Metrics.instance) {
+      Metrics.instance = this;
+    }
+
+    this.callDiagnosticMetrics = new CallDiagnosticMetrics();
+
+    // eslint-disable-next-line no-constructor-return
+    return Metrics.instance;
+  }
+
+  /**
+   * Initialize Call Diagnostic class
+   * @param meetingCollection
+   * @param webex
+   */
+  initialSetupCallDiagnosticMetrics(meetingCollection: any, webex: object) {
+    this.callDiagnosticMetrics.initialSetup(meetingCollection, webex);
+  }
+
+  /**
+   * Used for internal purposes only
+   * @param args
+   */
+  submitInternalEvent({
+    name,
+    payload,
+    options,
+  }: {
+    name: BehavioralEvent['name'];
```

Comment:
this should be internal event name not behavioral, same for payload

Reply:
oops, fixed